src/GafferScene/Unencapsulate.cpp

```diff
@@ -0,0 +1,287 @@
+}
+
+void Unencapsulate::hashBranchAttributes( const ScenePath &parentPath, const ScenePath &branchPath, const Gaffer::Context *context, IECore::MurmurHash &h ) const
+{
+	IECore::ConstObjectPtr capsuleObject = inPlug()->object( parentPath );
+	const Capsule *capsule = IECore::runTimeCast< const Capsule >( capsuleObject.get() );
+	h = capsule->scene()->attributesHash( concatScenePath( capsule->root(), branchPath ) );
+}
+
+IECore::ConstCompoundObjectPtr Unencapsulate::computeBranchAttributes( const ScenePath &parentPath, const ScenePath &branchPath, const Gaffer::Context *context ) const
+{
+	IECore::ConstObjectPtr capsuleObject = inPlug()->object( parentPath );
+	const Capsule *capsule = IECore::runTimeCast< const Capsule >( capsuleObject.get() );
+	return capsule->scene()->attributes( concatScenePath( capsule->root(), branchPath ) );
+}
+
+bool Unencapsulate::affectsBranchObject( const Gaffer::Plug *input ) const
+{
+	return input == inPlug()->objectPlug();
+}
+
+bool Unencapsulate::processesRootObject() const
+{
+    return true;
+}
+
+void Unencapsulate::hashBranchObject( const ScenePath &parentPath, const ScenePath &branchPath, const Gaffer::Context *context, IECore::MurmurHash &h ) const
+{
+	IECore::ConstObjectPtr o = inPlug()->object( parentPath );
+	const Capsule *capsule = IECore::runTimeCast< const Capsule >( o.get() );
+ 	if( branchPath.size() == 0 )
+	{
+		if( capsule )
+		{
+			h = IECore::NullObject::defaultNullObject()->Object::hash();
+		}
+		else
+		{
+			h = inPlug()->objectPlug()->hash();
+		}
+	}
+	else
+	{
+		h = capsule->scene()->objectHash( concatScenePath( capsule->root(), branchPath ) );
+	}
+}
+
+IECore::ConstObjectPtr Unencapsulate::computeBranchObject( const ScenePath &parentPath, const ScenePath &branchPath, const Gaffer::Context *context ) const
+{
+	IECore::ConstObjectPtr o = inPlug()->object( parentPath );
+	const Capsule *capsule = IECore::runTimeCast< const Capsule >( o.get() );
+ 	if( branchPath.size() == 0 )
+	{
+		// We're at a root location, not inside a capsule.  If there's a capsule here, we return a null
+		// object ( the capsule has now been expanded, so we don't want it in the scene any more ).
+		// If there's no capsule, we're not expanding anything, just pass through the source scene
+		if( capsule )
+		{
+			return IECore::NullObject::defaultNullObject();
+		}
+		else
+		{
+			return o;
+		}
+	}
+	return capsule->scene()->object( concatScenePath( capsule->root(), branchPath ) );
+}
+
+bool Unencapsulate::affectsBranchChildNames( const Gaffer::Plug *input ) const
+{
+	return input == inPlug()->objectPlug();
+}
+
+void Unencapsulate::hashBranchChildNames( const ScenePath &parentPath, const ScenePath &branchPath, const Gaffer::Context *context, IECore::MurmurHash &h ) const
+{
+	IECore::ConstObjectPtr capsuleObject = inPlug()->object( parentPath );
+	const Capsule *capsule = IECore::runTimeCast< const Capsule >( capsuleObject.get() );
+	if( !capsule )
+	{
+		h.append( 0 );
```

Comment:
Need to call `BranchCreator::hashBranchBound()` before calling `append()`. But I think we can use a little trick and do `h = outPlug()->childNamesPlug()->defaultValue()->Object::hash()` instead.

Reply:
Ah, you're right, that should be safe. I hadn't realized that unlike a lot of places where we deal with hashes, when hashMapping calls hashBranchChildNames, it passes in a temp hash, and then appends the temp hash to `h`, rather than just passing in `h`